lang/cs/Org.Apache.REEF.Common/Telemetry/CounterData.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Telemetry
+{
+    internal sealed class CounterData
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(CounterData)); 
+
+        /// <summary>
+        /// Registration of counters
+        /// </summary>
+        private readonly IDictionary<string, ICounter> _counters = new ConcurrentDictionary<string, ICounter>();
+
+        /// <summary>
+        /// Total increment since last sink for all the counters
+        /// </summary>
+        private readonly IDictionary<string, int> _totalIncrementSinceLastSink = new ConcurrentDictionary<string, int>();
```

Comment:
> private readonly IDictionary<string, int> _totalIncrementSinceLastSink = new ConcurrentDictionary<string, int> [](start = 7, length = 111)

I was actually thinking CounterData would wrap single counter and hold that counter  increment, so that we don't need to maintain separate dictionary :-)